crates/sparrow-compiler/src/ast_to_dfg.rs

```diff
@@ -95,6 +95,73 @@ pub(super) fn ast_to_dfg(
     )
 }
 
+pub fn add_udf_to_dfg(
+    udf: Arc<dyn Udf>,
+    dfg: &mut Dfg,
+    arguments: Resolved<Located<AstDfgRef>>,
+    data_context: &mut DataContext,
+    diagnostics: &mut DiagnosticCollector<'_>,
+) -> anyhow::Result<AstDfgRef> {
+    let argument_types = arguments.transform(|i| i.with_value(i.value_type().clone()));
+    let signature = udf.signature();
+
+    // TODO: No locations anymore
+    let fake_location = Located::new("fake".to_owned(), Location::internal_str("fake"));
+
+    // TODO: Ask about what python handles for us in regards to type checking, what we can pull
+    // out of here re: diagnostics since they're not reported(?)
```

Comment:
We currently have to put things in diagnostics in order for the python code to actually report an error. Ideally, this would get much simplified, and just return the (error or errors) from the builder method without first creating the diagnostic and then extracting the message.